### ISSUE TITLE
`gen_call_with_extracted_types_and_kwargs` doesn't always put the keyword arguments in the right place

### ISSUE THREAD

**Issue by vtjnash** [JuliaLang/julia#29735]
Note here that the `optimize=false` ended up in statement position, instead of being part of the call to `code_typed`. This gives the rather frustrating error "ERROR: syntax: invalid syntax optimize = false".

```julia
julia> Meta.@macroexpand @code_typed(optimize=false, cumsum(Real[1, 2.3]; dims=1))
quote
    #= /data/vtjnash/julia/usr/share/julia/stdlib/v1.1/InteractiveUtils/src/macros.jl:138 =#
    #28#results = begin
            #= /data/vtjnash/julia/usr/share/julia/stdlib/v1.1/InteractiveUtils/src/macros.jl:13 =#
            local #29#arg1 = cumsum
            #= /data/vtjnash/julia/usr/share/julia/stdlib/v1.1/InteractiveUtils/src/macros.jl:14 =#
            local (#30#args, #31#kwargs) = (InteractiveUtils.separate_kwargs)(Real[1, 2.3]; dims=1)
            #= /data/vtjnash/julia/usr/share/julia/stdlib/v1.1/InteractiveUtils/src/macros.jl:15 =#
            (InteractiveUtils.code_typed)((InteractiveUtils.Core).kwfunc(#29#arg1), (InteractiveUtils.Tuple){(InteractiveUtils.typeof)(#31#kwargs), (InteractiveUtils.Core).Typeof(#29#arg1), (InteractiveUtils.map)((InteractiveUtils.Core).Typeof, #30#args)...})
            optimize=false
        end
    #= /data/vtjnash/julia/usr/share/julia/stdlib/v1.1/InteractiveUtils/src/macros.jl:139 =#
    if (InteractiveUtils.length)(#28#results) == 1
        #28#results[1]
    else
        #28#results
    end
end
```

**Comment by Keno** [comment]
That's fixed in #29261

**Comment by Liozou** [comment]
Was fixed by #34367 I believe

**Comment by martinholters** [comment]
Looks like it.
